src/main/java/com/codereview/ipcounter/service/impl/UniqueIpCounter.java

```diff
@@ -0,0 +1,45 @@
+package com.codereview.ipcounter.service.impl;
+
+import com.codereview.ipcounter.service.IpCounter;
+import com.codereview.ipcounter.utils.ParserUtils;
+import java.net.UnknownHostException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class UniqueIpCounter implements IpCounter<String> {
+
+  private final int[] seen = new int[1 << 27];
```

Comment:
This looks very smart. I don't understand what's happening here. Add a comment, please

Reply:
`1 << 27` means 1 * 2^27
Add a comment with this value (134217728). Describe why did you pick that value exactly

Reply:
It is copy\paste solution from Internet. Bad approach. in any case Vsevolod is right and if you use some not clear code in your solution, you should describe it somehow.

Reply:
Also what does "seen" mean?